### PR TITLE
Remove `RefCell` usage from `ObligationForest`.

### PR DESCRIPTION
**Pull request by nnethercote**
It's not needed.

This doesn't affect performance, it just simplifies the code a little.

r? @nikomatsakis

### CODE REVIEW

File: src/librustc_data_structures/obligation_forest/mod.rs [hunk 1]
```diff
@@ -651,7 +651,7 @@ impl<O: ForestObligation> ObligationForest<O> {
         }
 
         node_rewrites.truncate(0);
-        self.node_rewrites.replace(node_rewrites);
+        std::mem::replace(&mut self.node_rewrites, node_rewrites);
```
Comment (nikomatsakis) [inline src/librustc_data_structures/obligation_forest/mod.rs]: Nit: Pre-existing, but why not just `self.node_writes = node_rewrites;`?

**Review by nikomatsakis — APPROVED** [review]
r=me modulo nit -- but feel free to ignore, I don't care

File: src/librustc_data_structures/obligation_forest/mod.rs (hunk 1 above)
Comment (nnethercote) [reply src/librustc_data_structures/obligation_forest/mod.rs]: Good point. I will update before landing.

**Comment by sinkuu** [comment]
Is there even the need for `mem::take`? `compress` is the only method that uses `self.node_rewrites`, and the method does not recurse. Unless I'm missing something, I think you can just use `self.node_rewrites` instead of moving it to a local and back.

**Comment by nnethercote** [comment]
> Unless I'm missing something, I think you can just use self.node_rewrites instead of moving it to a local and back.

I tried that, it leads to borrowck errors.

**Pull request merged**